akka-http-tests/src/test/scala/akka/http/scaladsl/OutgoingConnectionCancellationShouldKillEntityStreamSpec.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.{ Done, NotUsed }
+import akka.actor.ActorRef
+import akka.event.Logging
+import akka.http.scaladsl.client.RequestBuilding
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives
+import akka.http.scaladsl.settings.ConnectionPoolSettings
+import akka.stream._
+import akka.stream.scaladsl.{ Flow, Keep, Sink, Source }
+import akka.stream.stage.GraphStageLogic.StageActor
+import akka.stream.stage._
+import akka.testkit.{ AkkaSpec, ImplicitSender }
+import akka.util.ByteString
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future, Promise }
+import scala.util.{ Success, Try }
+
+/**
+ * Spec for https://github.com/akka/akka-http/issues/983
+ */
+class OutgoingConnectionCancellationShouldKillEntityStreamSpec extends AkkaSpec with ScalaFutures
```

Comment:
Please do not add one-off test setups. We have `LowLevelOutgoingConnectionSpec` for the low-level tests and `ConnectionPoolSpec` if you want to add a test that examines the pool behavior. These test files have all the necessary setup.

If you look through the tests in `LowLevelOutgoingConnectionSpec` there seem to be tests that test related behavior. It would be nice to understand what was going on when those tests were added.